pkg/internal/storage/repo/syncrepo/numeric_id.go

```diff
@@ -0,0 +1,78 @@
+package syncrepo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/storage/database/genquery"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/storage/database/models"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+// upsertNumericIDLookup inserts string IDs into the numeric ID lookup table to ensure each string ID has a corresponding numeric ID.
+// It executes custom INSERT ... SELECT ... ON CONFLICT DO NOTHING based on the result of the provided stringIDsQuery function.
+func upsertNumericIDLookup(ctx context.Context, db, tx *gorm.DB, stringIDsQuery func(db *gorm.DB) *gorm.DB) error {
+	dry := db.Session(&gorm.Session{DryRun: true, Initialized: true}) // NOTICE: Initialized to separate the dry run from the actual transaction (this makes the Session to clone the Statement)
+	query := stringIDsQuery(dry)
+
+	insertSelect := &gorm.Statement{DB: db}
+	clause.Expr{
+		SQL:  fmt.Sprintf("INSERT INTO %s (table_name, string_id) %s ON CONFLICT DO NOTHING", genquery.NumericIDLookup.TableName(), query.Statement.SQL.String()),
+		Vars: query.Statement.Vars,
+	}.Build(insertSelect)
```

Comment:
why not use `clause.Expr` for interpolation strings ? 
```suggestion
	clause.Expr{
	SQL: "INSERT INTO ?? (table_name, string_id) " + query.Statement.SQL.String() + " ON CONFLICT DO NOTHING",
	Vars: append(
		[]interface{}{clause.Table{Name: genquery.NumericIDLookup.TableName()}},
		query.Statement.Vars...,
	),
}.Build(insertSelect)
```

Reply:
1. I do use `Vars` to pass arguments to this expression
2. TableName is static (not provided from external)